python/lsst/ts/genericcamera/genericcameracsc.py

```diff
@@ -410,17 +437,102 @@ async def do_takeImages(self, id_data):
                     timestamp=time_stamp,
                     image_name=image_name,
                 )
-                await self.handle_exposure_saving(exposure, timestamp, image_name)
+                await self.camera.end_take_image()
+                await self.handle_exposure_saving(exposure, time_stamp, image_name)
 
-            await self.camera.end_take_image()
             await self.evt_endTakeImage.write()
+            # Adjust SEQNUM in case web service is unavailable
+            self.image_sequence_num += 1
         except Exception as e:
             self.log.exception(e)
             raise e
         finally:
             self.is_exposing = False
         self.log.info("takeImages - End")
 
+    def parse_key_value_map(self, key_value_map: str) -> None:
+        """Parse key/value map into additional keys and values.
+
+        Add extra information from the camera if necessary.
+
+        Parameters
+        ----------
+        key_value_map: `str`
+            The key/value map to parse.
+        """
+        try:
+            new_keyValueMap = ", ".join(
+                [
+                    key_value_map,
+                    self.camera.get_static_configuration_for_key_value_map(),
```

Comment:
So, if you agree to make `self.camera.get_static_configuration_for_key_value_map` return an empty string by default I think you can get away without this `try/except` clause. If you rather have it returning `None` I think it would be best to take that into account and handle the condition appropriately, instead of relying on the exception to do it.

Reply:
See comment above. Still have to do some check because of the dangling comma.